cmd/bootstrap/runner.go

```diff
@@ -186,3 +213,78 @@ func (r *runner) ensurePriorityClass(ctx context.Context, k8sClients k8sclient.I
 
 	return nil
 }
+
+func (r *runner) installOperators(ctx context.Context, helmClient helmclient.Interface) error {
+	var err error
+
+	operators := map[string]string{
+		"app-operator":   appOperatorVersion,
+		"chart-operator": chartOperatorVersion,
+	}
+
+	for name, version := range operators {
+		err = r.installOperator(ctx, helmClient, name, version)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	return nil
+}
+
+func (r *runner) installOperator(ctx context.Context, helmClient helmclient.Interface, name, version string) error {
+	var err error
+
+	var operatorTarballPath string
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("getting tarball URL for %#q", name))
+
+		operatorTarballURL, err := appcatalog.GetLatestChart(ctx, controlPlaneTestCatalogStorageURL, name, version)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("tarball URL is %#q", operatorTarballURL))
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", "pulling tarball")
+
+		operatorTarballPath, err = helmClient.PullChartTarball(ctx, operatorTarballURL)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("tarball path is %#q", operatorTarballPath))
+	}
+
+	{
+		defer func() {
+			fs := afero.NewOsFs()
+			err := fs.Remove(operatorTarballPath)
+			if err != nil {
+				r.logger.LogCtx(ctx, "level", "error", "message", fmt.Sprintf("deletion of %#q failed", operatorTarballPath), "stack", fmt.Sprintf("%#v", err))
+			}
+		}()
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("installing %#q", name))
+
+		// ReleaseName has unique suffix like in the control plane so the test
+		// app CRs need to use 0.0.0 for the version label.
+		opts := helmclient.InstallOptions{
+			ReleaseName: fmt.Sprintf("%s-unique", name),
+		}
+		err = helmClient.InstallReleaseFromTarball(ctx,
+			operatorTarballPath,
+			namespace,
+			nil,
+			opts)
+		if helmclient.IsReleaseAlreadyExists(err) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("%#q already installed", name))
+		} else if err != nil {
+			return microerror.Mask(err)
+		}
+
+		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("installed %#q", name))
```

Comment:
I would like to print `installed...` message only if this tool install operators, WDYT?
```suggestion
		} else {
		r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("installed %#q", name))
		}
```

Reply:
Good point. I did this slightly differently by returning early if the release already exists.